### PR TITLE
Improve handling of export errors

### PR DESCRIPTION
**Pull request by WCByrne**
Wrap native export calls in try catch and convert native exceptions to more helpful error messages.

Converting the native errors seemed to be the easiest way to handle this.

See: https://sketch-friends.slack.com/archives/C5YKGT26B/p1552063120097100

### CODE REVIEW

**Comment by bohemian-coding-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request may need a CHANGELOG entry.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  `Source/dom/export.js` changed, but not:
  - 📚 its docs

That's OK as long as you're refactoring.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Source/dom/export.js [hunk 1]
```diff
@@ -84,12 +102,15 @@ function exportToImageFile(nativeObjects, options) {
     options
   )
 
-  // export the pages
-  pages.forEach(exporter.exportPage)
-
-  // export the layers
-  if (layers.length) {
-    exporter.exportLayers(layers)
+  try {
```
Comment (mathieudutour) [inline Source/dom/export.js]: I'm not too sure about wrapping everything in try/catch. That breaks any optimizations the compiler can make.

I'd rather explicitly check for it:
```js
if (!nativeObjects.parentPage()) {
  throw error
}
```

Comment (WCByrne) [reply Source/dom/export.js]: I started going that route but then realized there are some edge cases in there. For example, some cases don't need a parent. This seemed less likely to cause conflicts if the internals were to change (except the error messages of course but that is covered by the tests). 

Any idea what that difference in optimizations would be? Significant for this?

**Comment by christianklotz** [comment]
Closing for now. May revisit at a later point.
